### DISCUSSION TITLE
development setup

### DISCUSSION THREAD

**Discussion by danish-mehmood** [makeplane/plane#1222] [General]
hello every one how do i setup my development environment , the contributing.md only mentions the project dependencies , is there another guide ?

**Comment by YourivdHeuvel** [comment]
Also running into this! Were you able to figure it out?:)

**Comment by thomasync** [comment]
I see that many are looking for a guide, I will make a PR with these steps in **CONTRIBUTING.md**

1. Clone the repo

```bash
git clone https://github.com/makeplane/plane
cd plane
chmod +x setup.sh
```

2. Run setup.sh

```bash
./setup.sh http://localhost
```

3. Define `NEXT_PUBLIC_API_BASE_URL=http://localhost` in **.env** file

```bash
echo "\nNEXT_PUBLIC_API_BASE_URL=http://localhost\n" >> .env
```

4. Run Docker compose up

```bash
docker compose up -d
```

5. Install dependencies

```bash
yarn install # bun install
```

6. Run the web app in development mode

```bash
yarn dev # bun run dev
```
